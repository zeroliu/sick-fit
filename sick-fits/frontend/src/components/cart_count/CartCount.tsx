import React from 'react';
import { CSSTransition, SwitchTransition } from 'react-transition-group';

import {
  Dot,
  ANIMATION_DURATION_MS,
  AnimatedCartCount,
} from './cart_count_styles';

interface Props {
  count: number;
}
export const CartCount: React.FC<Props> = ({ count }) => {
  return (
    <AnimatedCartCount>
      <SwitchTransition mode='out-in'>
        <CSSTransition
          key={count}
          timeout={ANIMATION_DURATION_MS}
          className='count'
          classNames='count'>
          <Dot>{count}</Dot>
        </CSSTransition>
      </SwitchTransition>
    </AnimatedCartCount>
  );
};
