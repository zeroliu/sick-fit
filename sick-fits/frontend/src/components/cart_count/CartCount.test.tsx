import { shallow } from 'enzyme';

import { CartCount } from './CartCount';

describe('<CartCount />', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<CartCount count={10} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('updates via props', () => {
    const wrapper = shallow(<CartCount count={10} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ count: 20 });
    expect(wrapper).toMatchSnapshot();
  });
});
