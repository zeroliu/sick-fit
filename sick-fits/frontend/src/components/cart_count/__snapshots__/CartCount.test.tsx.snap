// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> matches the snapshot 1`] = `
<cart_count_styles__AnimatedCartCount>
  <SwitchTransition
    mode="out-in"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={200}
    >
      <cart_count_styles__Dot>
        10
      </cart_count_styles__Dot>
    </CSSTransition>
  </SwitchTransition>
</cart_count_styles__AnimatedCartCount>
`;

exports[`<CartCount /> updates via props 1`] = `
<cart_count_styles__AnimatedCartCount>
  <SwitchTransition
    mode="out-in"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={200}
    >
      <cart_count_styles__Dot>
        10
      </cart_count_styles__Dot>
    </CSSTransition>
  </SwitchTransition>
</cart_count_styles__AnimatedCartCount>
`;

exports[`<CartCount /> updates via props 2`] = `
<cart_count_styles__AnimatedCartCount>
  <SwitchTransition
    mode="out-in"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="20"
      timeout={200}
    >
      <cart_count_styles__Dot>
        20
      </cart_count_styles__Dot>
    </CSSTransition>
  </SwitchTransition>
</cart_count_styles__AnimatedCartCount>
`;
